cmake/modules/SwiftSource.cmake

```diff
@@ -383,29 +390,66 @@ function(_compile_swift_files dependency_target_out_var_name)
     #
     # To work around this issue we touch the output files so that their mtime
     # always gets updated.
-    set(command_touch_outputs
-      COMMAND "${CMAKE_COMMAND}" -E touch ${outputs})
+    set(command_touch_standard_outputs
+      COMMAND "${CMAKE_COMMAND}" -E touch ${standard_outputs})
+    set(command_touch_apinotes_outputs
+      COMMAND "${CMAKE_COMMAND}" -E touch ${apinotes_outputs})
+    set(command_touch_module_outputs
+      COMMAND "${CMAKE_COMMAND}" -E touch ${module_outputs})
+  endif()
+
+  # First generate the api notes if we need them.
+  list(LENGTH command_create_apinotes num_apinotes_outputs)
+  if (NOT ${num_apinotes_outputs} EQUAL 0)
+    add_custom_command_target(
+        api_notes_dependency_target
+        ${command_create_dirs}
```

Comment:
Yep, thats a mistake.  Good catch.  I might just always have this target run, but in some cases it will only create directories and in others it will create directories and api notes.  If you have a preference for something else i'm also totally fine with alternatives.